crates/client/rpc/src/lib.rs

```diff
@@ -220,9 +226,97 @@ where
+        &self,
+        transaction_inputs: DeclareTransactionWithV0,
+    ) -> Result<(TransactionHash, ClassHash), StarknetRpcApiError> {
+        match transaction_inputs {
+            DeclareTransactionWithV0::V1(transaction) => {
+                self._declare_tx(transaction).await
+            }
+            DeclareTransactionWithV0::V0(declare_txn, program_vec, entrypoints, abi_length) => {
+                let txn_hash: TransactionHash = TransactionHash(StarkHash { 0: FieldElement::ONE.to_bytes_be() });
+
+                let program_decoded = Program::decode_all(&mut &*program_vec).unwrap();
+
+                let class_info;
+                let declare_transaction;
+
+                let try_class_info = ClassInfo::new(
+                    &blockifier::execution::contract_class::ContractClass::V0(ContractClassV0(Arc::from(
+                        ContractClassV0Inner { program: program_decoded, entry_points_by_type: entrypoints },
+                    ))),
+                    0,
+                    abi_length,
+                );
+
+                match try_class_info {
+                    Ok(val) => {
+                        class_info = val;
+                    }
+                    Err(_e) => return Err(StarknetRpcApiError::InternalServerError),
+                }
+
+                let try_declare_transaction = DeclareTransaction::new(
+                    starknet_api::transaction::DeclareTransaction::V0(declare_txn),
+                    txn_hash,
+                    class_info,
+                );
+
+                match try_declare_transaction {
+                    Ok(val) => {
+                        declare_transaction = val;
+                    }
+                    Err(e) => return Err(StarknetRpcApiError::from(e)),
+                }
+
+                self._declare_tx(declare_transaction).await
+            }
+        }
+    }
+
+    async fn _declare_tx(&self ,txn: DeclareTransaction) -> Result<(TransactionHash, ClassHash), StarknetRpcApiError> {
+        let best_block_hash = self.get_best_block_hash();
+        let current_block_hash = self.get_best_block_hash();
+        let contract_class = self
+            .overrides
+            .for_block_hash(self.client.as_ref(), current_block_hash)
+            .contract_class_by_class_hash(current_block_hash, txn.class_hash());
+
+        if let Some(contract_class) = contract_class {
+            error!("Contract class already exists: {:?}", contract_class);
+            return Err(StarknetRpcApiError::ClassAlreadyDeclared.into());
+        }
+
+        let extrinsic = self
+            .convert_tx_to_extrinsic(best_block_hash, AccountTransaction::Declare(txn.clone()))
+            .unwrap();
+
+        let res = submit_extrinsic(self.pool.clone(), best_block_hash, extrinsic).await;
+
+        match res {
+            Ok(_val) => Ok((txn.tx.compute_hash(Felt252Wrapper::from(self.chain_id().unwrap().0), false), txn.class_hash())),
```

Comment:
txn already has transaction_hash inside it right? we should call that and instead of hard coding it to ONE for V0 above, we can compute it there